test/jdk/com/sun/crypto/provider/KeyAgreement/SupportedDHParamGens.java

```diff
@@ -33,26 +34,53 @@
  * @run main/timeout=300 SupportedDHParamGens 832
  * @run main/timeout=300 SupportedDHParamGens 1024
  * @run main/timeout=600 SupportedDHParamGens 2048
+ * @run main/timeout=600 SupportedDHParamGens 3072
+ * @run main/timeout=600 SupportedDHParamGens 4096
  */
-
 import java.math.BigInteger;
 
 import java.security.*;
 import javax.crypto.*;
 import javax.crypto.interfaces.*;
 import javax.crypto.spec.*;
+import jdk.test.lib.security.DiffieHellmanGroup;
+import jdk.test.lib.security.SecurityUtils;
 
 public class SupportedDHParamGens {
 
+    static DiffieHellmanGroup dhGroup2048 = SecurityUtils.getTestDHGroup(2048);
+    static DiffieHellmanGroup dhGroup3072 = SecurityUtils.getTestDHGroup(3072);
+    static DiffieHellmanGroup dhGroup4096 = SecurityUtils.getTestDHGroup(4096);
+    static DHParameterSpec FFDHE2048_SPEC =  new DHParameterSpec(
+            dhGroup2048.getPrime(), dhGroup2048.getBase());
+
+    static DHParameterSpec FFDHE3072_SPEC =  new DHParameterSpec(
+            dhGroup3072.getPrime(), dhGroup3072.getBase());
+
+    static DHParameterSpec FFDHE4096_SPEC =  new DHParameterSpec(
+            dhGroup4096.getPrime(), dhGroup4096.getBase());
```

Comment:
It looks like the DiffieHelmanGroup fields are just used to intialize the DHParameterSpec. It might be cleaner to create the DHParameterSpec objects in a static initializer. 

Also, the DHParameterSpec objects can be `final`

Reply:
Actually, these vars are not needed, simplified with latest commit.